grafana/grafana
# Query History: Move filtering to service layer

**What this PR does / why we need it**:

This PR moves the logic for filtering query history items out of the component logic. We need this because when we enable remote query history all the filtering will happen in the backend.

There will be a small change in the behavior. In the multi-select to pick data sources, we will show all data sources, not only those that are in queries. This is because we will not have the list of data sources that have queries in the history when we switch to remote API.

**Which issue(s) this PR fixes**:



Fixes #44988

**Special notes for your reviewer**:

How to test it?
* Try to play with all filters (retention, data source selectors, search input and sorting)
* Play with different settings - selected retention period should be reflected as the maximum period when filtering. Previously used data source filters should be automatically put in the filter when the tab is opened.
